extensions/notebook/package.json

```diff
@@ -249,6 +249,18 @@
         "path": "./syntaxes/notebook.tmLanguage.json"
       }
     ],
+    "notebookExplorer": {
```

Comment:
We should figure out what we are trying to do here, half the code for notebooks is in core, half of it isn't and it isn't clear to me how we are deciding what goes where. Maybe we need a redesign doc around notebooks to properly handle the code structuring here.

Reply:
The 2 second version is that the notebooks extension contributes Jupyter-specific functionality (including books) as a notebook provider. We should have a doc anyway about what code lives where (and identify if there are any spots where code should be moved to/from core), but for this, we definitely want the extension to contribute the tree items; it can be an expensive operation if a folder is large.

Reply:
I mean for this specific change it doesn't really matter, but in general we should start being very hard about the different between our extension layer and our core layer, and less wishy washing about what goes where. If the notebooks extension is just the Jupiter kernel, we should probably just call it the jupyter kernel extension instead of notebooks.

If i delete the notebooks extension, will notebooks still work since we have a sql kernel? if not that tells me there is still an issue. (ex. if you delete the mssql extension, pgsql will still work in the query editor)

Reply:
The hope would be that the answer is "yes" for the core notebooks functionality, but I haven't played with it. Sounds like a good exercise to me.

Feedback heard about the naming convention.

Reply:
And sql is a little special since its built into core, a better example would be if we had a powershell kernel extension, could you still run a powershell notebook without the notebook extension